src/util.cpp

```diff
@@ -189,14 +189,27 @@ static void DebugPrintInit()
     vMsgsBeforeOpenLog = new std::list<std::string>;
 }
 
+/** Get the directory for log files (currently just debug.log) */
+static fs::path GetLogDir()
+{
+  if (gArgs.IsArgSet("-logdir")) {
+    fs::path logdirPath = fs::system_complete(gArgs.GetArg("-logdir", ""));
+    if (!logdirPath.empty()) {
+      fs::create_directories(logdirPath);
```

Comment:
Not sure if it should create directories,.. though if it does, in case the creation failed, it should not return an empty path, should rather return `GetDataDir()` in any failed case.

Reply:
I don't think the code (as written) can return an empty path. If `-logdir` is unset the outer if statement is skipped completely, the inner if statement is to handle the case where `-logdir` is set to the empty string. As I understand the boost::filesystem docs, if the `mkdir` fails then `fs::create_directories` will actually throw an exception.

I'm OK with removing the `fs::create_directories` call, the precedent I was following is that `GetDataDir()` creates its directory, although that's a bit different.

Reply:
Hmm.. not sure. I just found this in the docs: `Returns: true if a new directory was created, otherwise false.`

Reply:
As I understand the documentation: `fs::create_directories` returns true if it actually had to issue a `mkdir(2)` call, and returns false if it did not (because the directory already existed). If the directory needs to be created and the underlying `mkdir(2)` call failed, then an exception is raised.

Reply:
Oh.. yes.. your completely right. Nevermind then.